pkg/validation/cluster.go

```diff
@@ -896,6 +896,10 @@ func validateVSphereCloudSpec(spec *kubermaticv1.VSphereCloudSpec) error {
 		}
 	}
 
+	if spec.Networks != nil && spec.VMNetName != "" {
```

Comment:
I personally prefer using `len` because a slice can be non-nil but empty. In that case, this would fail. My opinion is that `len` is much safer option.

```suggestion
	if len(spec.Networks) != 0 && spec.VMNetName != "" {
```